### PR TITLE
Uprev to GHC 8.10.7

### PR DESCRIPTION
**Pull request by crogers1**
Refer to individual commits for more details. The primary PR can be found against xenclient-oe here:

https://github.com/OpenXT/xenclient-oe/pull/1469

### CODE REVIEW

**Comment by crogers1** [comment]
There was something strange with the commit history, I rebased on master and pushed to my branch.  Looks better now.

**Review by jandryuk — CHANGES_REQUESTED** [review]
Looks good.  Can the whitespace cleanup in https://github.com/OpenXT/manager/pull/207/commits/bd449d5ee4f7a101c0756c2370797b22f156f9cf just be folded into the ealier commits, please?

File: updatemgr/Db.hs [hunk 1]
```diff
@@ -52,12 +52,12 @@ type Path = String
...
-    (<*>) = ap
-
-instance (IsRemoteError e, MonadRpc e m) => Functor m where
-    fmap = ap . return
+--instance (IsRemoteError e, MonadRpc e m) => Applicative m where
+--    pure  = return
+--    (<*>) = ap
+--
+--instance (IsRemoteError e, MonadRpc e m) => Functor m where
+--    fmap = ap . return
```
[the remark this answers is not in the document]
Comment (crogers1) [reply updatemgr/Db.hs:60]: They can be deleted. oversight during cleanup.

File: rpc-proxy/Msg/JsonConv.hs [hunk 2]
```diff
@@ -69,19 +85,19 @@ dtypes GuessTypes as = map dtype as
 convToMethodCall :: TypeConvMode -> JReq -> Maybe MethodCall
 convToMethodCall sig r
   = MethodCall
-      <$> parseObjectPath (jreqPath r)
+      <$> parseObjectPathInternal (jreqPath r)
       <*> pure (parseInterfaceName $ jreqInterface r)
       <*> parseMemberName (jreqMethod r)
-      <*> pure (parseBusName $ jreqDest r)
       <*> pure Nothing
+      <*> pure (parseBusName $ jreqDest r)
```
[the remark this answers is not in the document]
Comment (crogers1) [reply rpc-proxy/Msg/JsonConv.hs]: Split this change out specifically and put it into the "Update conversion methods" commit, which has a new sha after the rebase and push [1a82888](https://github.com/OpenXT/manager/pull/207/commits/1a828883ac2597ac6bf4a063f0d9d5d812c3bd47).

File: rpc-proxy/Msg/JsonConv.hs [hunk 3]
```diff
@@ -187,11 +209,197 @@ convVariant v = go (variantType v) where
...
+-- Below this line, reimplement parsing logic from upstream
+-- using "internal" function name to differentiate between
+-- the two. We don't want to execute within a MonadThrow,
+-- just continue using the Maybe logic that the original
+-- rpc-proxy code was based off when it imported the now deprecated
+-- dbus-core lib. parseSigFullInternal worries about catching
+-- any throws and converts them to a 'Nothing'.
+parseObjectPathInternal :: String -> Maybe ObjectPath
+parseObjectPathInternal s = do
+    maybeParseString parserObjectPathInternal s
```
[the remark this answers is not in the document]
Comment (crogers1) [reply rpc-proxy/Msg/JsonConv.hs]: The "Internal" forward ports have been removed completely.

[the remark this answers is not in the document]
Comment (crogers1) [reply rpc-proxy/Msg/JsonConv.hs]: Removed the removal commit, and just dropped all the Internal forward port stuff in the commit that originally introduced it.

**Comment by crogers1** [comment]
Was able to consolidate a few more commits.  Took the Setup.hs changes and put them all into a single commit, no need for the intermediate change to Setup.hs.in since we just want the symlinks back to Setup.hs.  Also squashed the single line change to the `inparams` for Msg/Json.hs that was it's own commit and put that into the larger commit for that module [a9a1bb6](https://github.com/OpenXT/manager/pull/207/commits/a9a1bb69ac54c5e60ce84d3ee825d0ffc732a18e)

**Comment by crogers1** [comment]
Yea absolutely.

**Review by jandryuk — APPROVED** [review]
Thanks!

**Comment by jandryuk** [comment]
build failure:
| 
| XenMgr/Host.hs:816:13: error:
|     • Variable not in scope: catch :: IO String -> (p0 -> m0 a1) -> t
|     • Perhaps you meant one of these:
|         ‘E.catch’ (imported from Control.Exception),
|         ‘match’ (imported from Text.Regex.Posix)
|     |
| 816 |   pcmDevs = catch ( readFile "/proc/asound/pcm" )
|     |    

From after the audio changes, I guess

**Comment by jandryuk** [comment]
Tabs slipped into:
xenmgr/Vm/Config.hs (sorry about that)
xenmgr/Vm/PciDatabase.hs

Fixup in https://github.com/OpenXT/manager/pull/207/commits/2e257d980b05c0183ca24055f3122ec0a930eb76 please

There is another catch error in XenMgr/Expose/HostObject.hs:402:11

**Comment by crogers1** [comment]
Fixed the tab in Config, don't see any tabs in PciDatabase that aren't commented out.  Did not observe any build errors in HostObject.hs

**Comment by jandryuk** [comment]
Oh, I didn't realize PciDatabase's hard tabs were commented out - I just saw them in the grep output.

Thanks for the fixup.

Did you build this branch in isolation or merged into master?  I think the compile errors only show up with it merged into master.

**Comment by crogers1** [comment]
it was in isolation, I'll have to rebase it on master

**Pull request merged**